File: src/main/java/guru/xgm/angelscript/openapi/fileTypes/ZincFileType.java
package guru.xgm.angelscript.openapi.fileTypes;

import com.intellij.openapi.fileTypes.LanguageFileType;
import guru.xgm.angelscript.icons.ZincIcons;
import guru.xgm.angelscript.lang.ZincLanguage;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public final class ZincFileType extends LanguageFileType {

    public static final ZincFileType INSTANCE = new ZincFileType();

    private ZincFileType() {
        super(ZincLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "ZINC";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "ZINC language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "zn";
    }

    @Override
    public @NotNull Icon getIcon() {
        return ZincIcons.FILE;
    }
}

File: src/main/kotlin/raft/war/language/zinc/openapi/fileTypes/ZincFileType.kt
package raft.war.language.zinc.openapi.fileTypes

import com.intellij.openapi.fileTypes.LanguageFileType
import raft.war.language.zinc.icons.ZincIcons
import raft.war.language.zinc.lang.ZincLanguage
import javax.swing.Icon

class ZincFileType private constructor() : LanguageFileType(ZincLanguage.instance) {
    override fun getName(): String = ZincLanguage.NAME

    override fun getDescription(): String = ZincLanguage.NAME

    override fun getDefaultExtension(): String = EXTENSION

    override fun getIcon(): Icon = ZincIcons.FILE

    @Suppress("CompanionObjectInExtension")
    companion object {
        val instance: ZincFileType = ZincFileType()
        const val EXTENSION = "zn"
    }
}
